File: shapetrees-java-core/src/main/java/com/janeirodigital/shapetrees/core/methodhandlers/AbstractValidatingMethodHandler.java
package com.janeirodigital.shapetrees.core.methodhandlers;

import com.janeirodigital.shapetrees.core.ResourceAccessor;
import com.janeirodigital.shapetrees.core.ShapeTreeRequestHandler;
import lombok.extern.slf4j.Slf4j;

/**
 * Abstract class providing reusable functionality to different method handlers
 */
@Slf4j
public abstract class AbstractValidatingMethodHandler {
    private static final String DELETE = "DELETE";
    protected final ResourceAccessor resourceAccessor;
    protected final ShapeTreeRequestHandler requestHandler;

    protected AbstractValidatingMethodHandler(ResourceAccessor resourceAccessor) {
        this.resourceAccessor = resourceAccessor;
        this.requestHandler = new ShapeTreeRequestHandler(resourceAccessor);
    }
}

File: asTypescript/packages/core/src/methodhandlers/AbstractValidatingMethodHandler.ts
// Corresponding shapetrees-java package: com.janeirodigital.shapetrees.core.methodhandlers
import { ResourceAccessor } from '../ResourceAccessor';
import { ShapeTreeRequestHandler } from '../ShapeTreeRequestHandler';
import * as Slf4j from 'lombok/extern/slf4j';

/**
 * Abstract class providing reusable functionality to different method handlers
 */
@Slf4j
export abstract class AbstractValidatingMethodHandler {

   private static readonly DELETE: string = "DELETE";

   protected readonly resourceAccessor: ResourceAccessor;

   protected readonly requestHandler: ShapeTreeRequestHandler;

  protected constructor(resourceAccessor: ResourceAccessor) {
    this.resourceAccessor = resourceAccessor;
    this.requestHandler = new ShapeTreeRequestHandler(resourceAccessor);
  }
}
